backend/src/resolvers.js

```diff
@@ -1,60 +1,92 @@
-const { UserInputError } = require("apollo-server");
+import { UserInputError } from "apollo-server";
+import bcrypt from "bcrypt";
 
 const resolvers = {
     Query: {
-        users: (parent, args, context) => context.dataSources.db.allUsers(),
-        posts: (parent, args, context) => context.dataSources.db.allPosts(),
+        users: (_, args, context) => context.dataSources.db.allUsers(),
+        posts: (_, args, context) => context.dataSources.db.allPosts(),
     },
     Mutation: {
-        write: async (parent, args, context) => {
-            const newPost = {
-                title: args.post.title,
-                authorName: args.post.author.name,
-            };
-            const author = await context.dataSources.db.getUser(newPost.authorName);
-            if (!author) {
-                throw new UserInputError("Invalid user", { invalidArgs: newPost.authorName });
+        signup: async (_, args, { dataSources }) => {
+            const name = args.name;
+            const email = args.email;
+            const password = args.password;
+            if (password.length < 9) {
+                throw new UserInputError("Password is too short", { invalidArgs: password });
             }
```

Comment:
Not exactly according to spec, but fully OK to me:
>Accept only passwords with a length of at least 8 characters.

One
>:star: For password validation.

Reply:
Agree to both. As this only makes it more secure, it's okay I guess.